CMakeLists.txt

```diff
@@ -12,7 +12,15 @@ project(bpfilter
 include(GNUInstallDirs)
 
 find_package(Doxygen REQUIRED)
-find_package(cmocka REQUIRED)
+
+find_package(PkgConfig)
+pkg_check_modules(bpf REQUIRED IMPORTED_TARGET libbpf)
+pkg_check_modules(elf REQUIRED IMPORTED_TARGET libelf)
+pkg_check_modules(cmocka REQUIRED IMPORTED_TARGET cmocka)
+
+# Required to get CMake to pass PIE flags to the compiler/linker
+include(CheckPIESupported)
+check_pie_supported()
```

Comment:
`POSITION_INDEPENDENT_CODE` property is not supported if this function is not called? Does it fails the generation if PIE is not supported?

Reply:
In this case, you should probably add a warning if `CMAKE_C_LINK_PIE_SUPPORTED` is false.